packages/app/src/state/staking/actions.ts

```diff
@@ -108,6 +142,35 @@ export const fetchEscrowData = createAsyncThunk<
 	}
 })
 
+export const fetchEscrowV2Data = createAsyncThunk<
+	{ escrowData: EscrowData<string>[]; totalVestable: string },
+	void,
+	ThunkConfig
+>('staking/fetchEscrowV2Data', async (_, { extra: { sdk } }) => {
+	const { escrowData, totalVestable } = await sdk.kwentaToken.getEscrowV2Data()
+
+	return {
+		escrowData: escrowData.map((e) => ({
+			...e,
+			vestable: e.vestable.toString(),
+			amount: e.amount.toString(),
+			fee: e.fee.toString(),
+		})),
+		totalVestable: totalVestable.toString(),
+	}
+})
+
+export const fetchStakeMigrateData = createAsyncThunk<void, void, ThunkConfig>(
+	'stakeMigrateData/fetch',
+	async (_, { dispatch }) => {
+		await dispatch(fetchStakingData())
```

Comment:
I don't think you need to await these unless there is some reason for them to be synchronous?

Reply:
Yes. Fixed.